pkg/resources/stream_common.go

```diff
@@ -196,19 +196,24 @@ func handleStreamRead(d *schema.ResourceData,
 	stream *sdk.Stream,
 	streamDescription *sdk.Stream,
 ) error {
-	if err := d.Set("comment", stream.Comment); err != nil {
-		return err
-	}
+	return errors.Join(
+		d.Set("comment", stream.Comment),
+		d.Set(ShowOutputAttributeName, []map[string]any{schemas.StreamToSchema(stream)}),
+		d.Set(DescribeOutputAttributeName, []map[string]any{schemas.StreamDescriptionToSchema(*streamDescription)}),
+		d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()),
+	)
+}
 
-	if err := d.Set(ShowOutputAttributeName, []map[string]any{schemas.StreamToSchema(stream)}); err != nil {
-		return err
-	}
-	if err := d.Set(DescribeOutputAttributeName, []map[string]any{schemas.StreamDescriptionToSchema(*streamDescription)}); err != nil {
-		return err
-	}
-	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
-		return err
+func isStale(d *schema.ResourceData) bool {
```

Comment:
Question: What does "stale" mean in the context of a stream, and why do we want to recreate it when it's true?

Reply:
You can read more [here](https://docs.snowflake.com/en/user-guide/streams-intro#data-retention-period-and-staleness).

When a stream is stale, and you try reading from it, you get an error message telling you to recreate it ([link](https://community.snowflake.com/s/article/using-tasks-to-avoid-stale-streams-when-incoming-data-is-empty)). This logic is also requested by the community - see https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1150. I added a comment to make it more obvious.

Reply:
Got it, can we put the first link to the description of the stale field then?